src/board/helpers.mjs

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable guard-for-in */
+/* eslint-disable no-restricted-syntax */
+
+// contains functions to duplicate, flatten, unflatten, and shuffle the grid
+
+function duplicateGrid(grid) {
+  const result = [];
+  grid.forEach((row, gridIndex) => {
+    const newRow = [];
+    row.forEach((color, rowIndex) => {
+      newRow[rowIndex] = color;
+    });
+    result[gridIndex] = newRow;
+  });
+  return result;
+}
+
+function flattenGrid(grid) {
+  const result = [];
+  grid.forEach((row) => {
+    row.forEach((color) => {
+      result.push(color);
+    });
+  });
+  return result;
+}
+
+function shuffle(arr) { // Fisher-Yates shuffle
+  for (let i = arr.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+    // TODO: ask about this
```

Comment:
is this comment still needed?

Reply:
I'm going to ask someone about this, it works just fine, but I want to avoid disabling eslint rules when possible.

Reply:
Kind of on the fence still about disabling eslint rules and trying not to do it too much, but removed this comment for now.

Reply:
i was wondering about the `TODO: ask about this comment`, not sure if it looks like i was asking about an eslint comment